models/contacts.js

```diff
@@ -1,19 +1,102 @@
-// const fs = require('fs/promises')
+const fs = require("fs/promises");
+const { nanoid } = require("nanoid");
 
-const listContacts = async () => {}
+const contacts = fs.readFile("models/contacts.json");
 
-const getContactById = async (contactId) => {}
+const listContacts = async () => {
+  try {
+    return JSON.parse(await contacts);
+  } catch (error) {
+    console.log(error);
+  }
+};
 
-const removeContact = async (contactId) => {}
+const getContactById = async (contactId) => {
+  try {
+    const parsedData = JSON.parse(await contacts);
 
-const addContact = async (body) => {}
+    return parsedData.find((data) => data.id === contactId);
+  } catch (error) {
+    console.log(error);
+  }
+};
 
-const updateContact = async (contactId, body) => {}
+const removeContact = async (contactId) => {
+  try {
+    const parsedData = JSON.parse(await contacts);
+    const checkIfExist = parsedData.find((data) => data.id === contactId);
+
+    if (checkIfExist) {
+      const contactsAfterRemove = parsedData.filter(
+        (contact) => contact.id !== contactId
+      );
+      fs.writeFile("models/contacts.json", JSON.stringify(contactsAfterRemove));
```

Comment:
you can create method for storing users in database ( currently json file ). Just extract this to separated method with contacts as parameter
`fs.writeFile("models/contacts.json", JSON.stringify(contactsAfterRemove))`

Reply:
donee